### ISSUE TITLE
Fix and Update Dependencies and Package Workflow

### ISSUE THREAD

**Issue by s0v1x** [s0v1x/StocksMA#12]
This issue focuses on fixing and updating dependencies and the package workflow. It has been identified that there are several vulnerabilities and issues in the current dependencies and package setup.

Moreover, improvements to the package workflow should be implemented to enhance overall security and stability.

**Comment by s0v1x** [comment]
@Ahmed-Z-D Could you please review these?

**Comment by Ahmed-Z-D** [comment]
Hi @s0v1x 

The dependencies of the package have been updated and resolved. As for the proposed package workflow improvement, do you have any suggestions that you would like to implement, apart from unit testing?

**Comment by iamleblanc** [comment]
@Ahmed-Z-D Thanks for the efforts. We should definitely have a discussion about this matter since I don't have a specific design or new additions in mind at the moment. However, in the meantime, could you please release a new package version for PyPI? We have made numerous changes that need to be included.

P.S. Could you please send me your email privately?

**Comment by Ahmed-Z-D** [comment]
Absolutely! I'm on board with having a thorough discussion about the various aspects of this package. By the way, as soon as I tackle issue #14, I'll happily proceed with your request to release a fresh version of the package on PyPI.
